OneSignalSDK/onesignal/src/main/java/com/onesignal/PermissionsActivity.java

```diff
@@ -104,10 +109,16 @@ public void onRequestPermissionsResult(final int requestCode, @NonNull String pe
             public void run() {
                boolean granted = grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED;
                LocationGMS.sendAndClearPromptHandlers(true, granted);
-               if (granted)
+               if (granted) {
                   LocationGMS.startGetLocation();
-               else
+               } else {
+                  if (fallbackToSettings &&
+                          neverAskAgainClicked &&
+                          !ActivityCompat.shouldShowRequestPermissionRationale(PermissionsActivity.this, LocationGMS.requestPermission)) {
```

Comment:
Maybe we should move this if check into a separate method to clean up this long `if-statement`, something like below?
```
if (shouldShowLocationPermissionSettings)
    showLocationPermissionSettings();
```
**OR**
```
private void attemptToShowLocationPermissionSettings() {
    if (fallbackToSettings 
            && neverAskAgainClicked 
            && !ActivityCompat.shouldShowRequestPermissionRationale(PermissionsActivity.this, LocationGMS.requestPermission))
        showLocationPermissionSettings();
}
```